pkg/reconciler/pipelinerun/affinity_assistant_test.go

```diff
@@ -159,12 +161,16 @@ func TestCreateAndDeleteOfAffinityAssistantPerPipelineRun(t *testing.T) {
 	}, {
 		name:                  "other Workspace type",
 		pr:                    testPRWithEmptyDir,
-		expectStatefulSetSpec: nil,
+		expectStatefulSetSpec: &appsv1.StatefulSetSpec{},
```

Comment:
nit: It doesn't really make sense to differentiate between nil and a pointer to an empty struct; I'd have tests treat them as equivalent rather than asserting a specific behavior.

Reply:
Prior to this commit, we don't create `AffinityAssistant` for `TaskRuns` without a `pvc` based workspace in `coschedule per pipelinerun` mode, the `statefulset` is NOT expected to be created (`nil`).

In this commit, we create `AffinityAssistant` for all `TaskRuns`, so we do expect an `AffinityAssitant` is created (but there is no `pvc` or `volumeclaimtemplate` in the `StatefulSetSpec`)

Reply:
I'm a bit confused, why would we want to validate that a statefulset is created with an empty spec? how is that different from a nil pointer?

Reply:
Sorry for the confusion. The confusing point is that we had a `StatefulSetSpec` filter here: https://github.com/tektoncd/pipeline/blob/53e71bf8a014b6bb9174b8496bc3a829c700dd60/pkg/reconciler/pipelinerun/affinity_assistant_test.go#L1218, which ignores the `Replica` and `Selector` before.

In the test case, it is actually a spec with `Replica` and `Selector` but not an empty spec (we didn't have it before since they will not be validated anyways due to the filter). I think the best way address this confusion is to remote this filter and validate the selector and replicas (and we get more validated coverage 😄 )